### PR TITLE
Create points_selection_vispy.py

### PR DESCRIPTION
**Pull request by PennyQ**
Free lasso & shape (rectangle and ellipse) selection as well as 'picking' for 3d points data, switched by keypress 1-4, using QT as display window. There is a bug of setting an empty input for the drawing line visual should be fixed later.

### CODE REVIEW

File: points_selection_vispy.py [hunk 1]
```diff
@@ -0,0 +1,266 @@
+"""
+    3D Points Picking & Selection tool based on Vispy library
+    Author: Penny Xuran Qian (NAOC/SAO) pennyqxr@gmail.com
+    Date: Feb 15, 2016
+    License: GPL
```
Comment (astrofrog) [inline points_selection_vispy.py]: @PennyQ note that Glue and the 3D viewers are released under the BSD license, so if you commit this code under the GPL, we won't be able to use it (GPL can't be used in BSD package). Could you update to BSD?

File: points_selection_vispy.py [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+    corner3[:, 0] = center[0] + bias1[2]
+    corner3[:, 1] = center[1] + bias2[2]
+    corner3[:, 2] = 0
+
+
+    corner4[:, 0] = center[0] - bias1[3]
+    corner4[:, 1] = center[1] + bias2[3]
+    corner4[:, 2] = 0
+
+    output = np.concatenate(([[center[0], center[1], 0.]],
```
Comment (astrofrog) [inline points_selection_vispy.py]: Could you add a comment to explain what 'output' is an array of?

File: points_selection_vispy.py [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+                             corner1,
+                             [[center[0], center[1] - half_height, 0.]],
+                             corner2,
+                             [[center[0] + half_width, center[1], 0.]],
+                             corner3,
+                             [[center[0], center[1] + half_height, 0.]],
+                             corner4,
+                             [[center[0] - half_width, center[1], 0.]]))
+
+    vertices = np.array(output, dtype=np.float32)
```
Comment (astrofrog) [inline points_selection_vispy.py]: ``output`` is already an array, so you can do:

```python
vertices = output.astype(np.float32)
```

Comment (astrofrog) [inline points_selection_vispy.py]: Could you add a comment explaining why you are converting to 32-bit floats?

**Comment by astrofrog** [comment]
Looks good apart from small comments above! I would also suggest splitting the file into two - one with the code that includes the selection classes and the callback functions, and one with the actual example. This will then allow us to use this framework for other examples too and in the real 3d plugin.

**Comment by PennyQ** [comment]
@astrofrog Thanks for you advice, they are very useful. 
I agree with having an individual 'SelectionVisual' to take care of drawing selection shape and highlighting the selected part. But I found it's harder than I expected because the `selection shape` will be displayed on the `scene' rather than 'view' to avoid the transform in 3D coordinate. I haven't got an idea on this yet, do you have any ideas?

**Comment by astrofrog** [comment]
@PennyQ - just to clarify, I'm not suggesting that you would have a ``SelectionVisual``, but I was more advocating just splitting the source code into two files (but you can keep the current structure of the code)

**Comment by PennyQ** [comment]
@astrofrog [Here is the two scripts](https://github.com/glue-viz/glue-3d-selection/tree/master/scatter_selection)  as `selection_common.py` is for drawing selection line and realizing selection functionality, the `selection_widget.py` is the widget to display. We could close these three issues if you think the scripts are ok :)

**Pull request merged**